plugin/src/main/java/org/opensearch/ml/model/MLModelGroupManager.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.model;
+
+import static org.opensearch.ml.common.CommonValue.ML_MODEL_GROUP_INDEX;
+
+import java.time.Instant;
+import java.util.HashSet;
+
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.index.IndexRequest;
+import org.opensearch.action.search.SearchRequest;
+import org.opensearch.action.search.SearchResponse;
+import org.opensearch.action.support.WriteRequest;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.util.CollectionUtils;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.common.xcontent.XContentType;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.core.xcontent.ToXContent;
+import org.opensearch.core.xcontent.XContentBuilder;
+import org.opensearch.index.query.BoolQueryBuilder;
+import org.opensearch.index.query.TermQueryBuilder;
+import org.opensearch.ml.common.AccessMode;
+import org.opensearch.ml.common.MLModelGroup;
+import org.opensearch.ml.common.transport.model_group.MLRegisterModelGroupInput;
+import org.opensearch.ml.helper.ModelAccessControlHelper;
+import org.opensearch.ml.indices.MLIndicesHandler;
+import org.opensearch.ml.utils.RestActionUtils;
+import org.opensearch.search.builder.SearchSourceBuilder;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class MLModelGroupManager {
+    private final MLIndicesHandler mlIndicesHandler;
+    private final Client client;
+    ClusterService clusterService;
+
+    ModelAccessControlHelper modelAccessControlHelper;
+
+    @Inject
+    public MLModelGroupManager(
+        MLIndicesHandler mlIndicesHandler,
+        Client client,
+        ClusterService clusterService,
+        ModelAccessControlHelper modelAccessControlHelper
+    ) {
+        this.mlIndicesHandler = mlIndicesHandler;
+        this.client = client;
+        this.clusterService = clusterService;
+        this.modelAccessControlHelper = modelAccessControlHelper;
+    }
+
+    public void createModelGroup(MLRegisterModelGroupInput input, ActionListener<String> listener) {
+        try {
+            String modelName = input.getName();
+            User user = RestActionUtils.getUserContext(client);
+            try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                validateUniqueModelGroupName(input.getName(), ActionListener.wrap(modelGroups -> {
```

Comment:
What about here? Is there any chance of facing index not found exception?

Reply:
Found nothing of sort while testing. Maybe I can do more testing and confirm

Reply:
May be delete the index and then try?